### ISSUE TITLE
Feature: continuous voice recognition

### ISSUE THREAD

**Issue by daniel-luper** [CS-Journeys/hands-free-chess#34]
### Context
There is a non-negligible time interval during which the user's voice is not recorded. Either the user has to wait until the microphone gets reactivated (but paying attention to the timing is a pain), or, if they don't wait, they risk not being heard by HFC. 

### Current behavior
1. Microphone starts listening
2. Microphone detects noise (the user said something or there's some other noise from the environment)
3. Microphone stops whenever noise stops
4. `command_recognition.py` sends a request to the Google Speech API
5. HFC waits for a response...
6. `command_recognition.py` processes the response
7. If the command is a move command (as opposed to "pause", "exit", etc.), then `board_recognition.py` searches for a chessboard on the screen
8. If the move is a legal move, then `mouse_controller.py` moves the piece with the mouse
9. FINALLY, the microphone starts listening again lol

#### THREAD A:
1. Microphone starts listening
2. Microphone detects noise
3. Microphone stops whenever noise stops
4. Store the recorded audio in a buffer that is shared with thread B
5. Return to step 1

#### THREAD B:
1. Combine the most recent recordings (in case the microphone recording stopped and restarted in the middle of a command)
2. Send a request to the Google Speech API
3. Wait for a response
4. Store the response in a buffer that is shared with thread C
5. Return to step 1

#### THREAD C:
1. If the oldest response is not a recognized command, delete it and repeat step 1 to process the next response
2. If the response is ['pause'], ['exit'], etc., then handle that and return to step 1
3. If the response is a move command, then search for a chessboard on the screen
4. If the move is a legal, then move the piece with the mouse
5. Return to step 1

### Alternative solutions
- Simply put the recording and/or API request on a separate thread. Not perfect, but probably better than the current behavior
- Use an offline speech recognition API to reduce request/response wait time ([CMU Sphinx](https://cmusphinx.github.io/wiki/)?)
- Use another API or library specifically for continuous/asynchronous speech recognition
- Build your own speech recognition artificial intelligence
- Some combination of the previously mentioned solutions
- Something totally different

### Milestone???
This is a really helpful feature but it might take too much time for v1.0. Maybe we could release it in v1.1? or v2.0? Even though it's somewhat difficult, maybe it still needs to be released it v1.0 because it's so important. What do you think?
